.github/workflows/e2e-upgrade.yml

```diff
@@ -0,0 +1,183 @@
+      id-token: write
+      contents: read
+    steps:
+      - name: Check out repository
+        uses: actions/checkout@755da8c3cf115ac066823e79a1e1788f8940201b # v3.2.0
+        with:
+          fetch-depth: 0
+          ref: ${{ !github.event.pull_request.head.repo.fork && github.head_ref || '' }}
+
+      - name: Setup Go environment
+        uses: actions/setup-go@6edd4406fa81c3da01a34fa6f6343087c207a568 # v3.5.0
+        with:
+          go-version: "1.19.5"
+
+      - name: Login to Azure
+        if: matrix.provider == 'azure'
+        uses: ./.github/actions/login_azure
+        with:
+          azure_credentials: ${{ secrets.AZURE_E2E_CREDENTIALS }}
+
+      - name: Create Azure resource group
+        if: matrix.provider == 'azure'
+        id: az_resource_group_gen
+        shell: bash
+        run: |
+          uuid=$(cat /proc/sys/kernel/random/uuid)
+          name=e2e-test-${uuid%%-*}
+          az group create --location northeurope --name "$name" --tags e2e
+          echo "res_group_name=$name" >> "$GITHUB_OUTPUT"
+
+      - name: Create cluster with 'fromVersion' CLI.
+        id: e2e_test
+        uses: ./.github/actions/e2e_test
+        with:
+          workerNodesCount: "2"
+          controlNodesCount: "3"
+          cloudProvider: ${{ matrix.provider }}
+          osImage: ${{ inputs.fromVersion }}
+          cliVersion: ${{ inputs.fromVersion }}
+          isDebugImage: "false"
+          azureSubscription: ${{ secrets.AZURE_E2E_SUBSCRIPTION_ID }}
+          azureTenant: ${{ secrets.AZURE_E2E_TENANT_ID }}
+          azureClientID: ${{ secrets.AZURE_E2E_CLIENT_ID }}
+          azureClientSecret: ${{ secrets.AZURE_E2E_CLIENT_SECRET }}
+          azureUserAssignedIdentity: ${{ secrets.AZURE_E2E_USER_ASSIGNED_IDENTITY }}
+          azureResourceGroup: ${{ steps.az_resource_group_gen.outputs.res_group_name }}
+          gcpProject: ${{ secrets.GCP_E2E_PROJECT }}
+          gcp_service_account: "constellation-e2e@constellation-331613.iam.gserviceaccount.com"
+          gcpClusterServiceAccountKey: ${{ secrets.GCP_CLUSTER_SERVICE_ACCOUNT }}
+          test: "nop"
+
+      - name: Build CLI
+        if: inputs.toVersion == ''
+        uses: ./.github/actions/build_cli
+
+      - name: Fetch CLI
+        if: inputs.toVersion != ''
+        shell: bash
+        run: |
+          curl -fsSL https://github.com/edgelesssys/constellation/releases/download/${{ inputs.toVersion }}/constellation-linux-amd64 > constellation
+          chmod u+x constellation
+          "$(pwd)" >> "$GITHUB_PATH"
+          export PATH="$PATH:$PWD"
+          constellation version
+          # Do not spam license server from pipeline
+          sudo sh -c 'echo "127.0.0.1 license.confidential.cloud" >> /etc/hosts'
+
+      - name: Find latest nightly image
+        id: find-image
+        if: inputs.toImage == ''
+        uses: ./.github/actions/versionsapi
+        with:
+          command: latest
+          ref: main
+          stream: nightly
```

Comment:
Could you add a test after this step that checks that the located image is not older than x hrs? Maybe 2hrs or sth. I already see myself wondering why the upgrade failed only to realize that I didn't build an image with an up2date bootstrapper. 

Alternatively we could trigger the image build here. Or use debug images. Not sure what the best approach would be.